src/debuggability.js

```diff
@@ -161,6 +163,46 @@ Promise.prototype._propagateFrom = function (parent, flags) {
     USE(flags);
 };
 
+Promise.monitor = function () {
+    // Pending promises are stored in array for better performance: number
+    // of simultaneously pending promises is assumed to be relatively small,
```

Comment:
Cant rely on ES6,just put a unique auto-incremented id on each promise and use that in as an object key